app/main.py

```diff
@@ -1,2 +1,51 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.table = [[] for _ in range(self.capacity)]
+        self.length = 0
+        self.threshold = int(self.capacity * self.load_factor)
+
+    def __getitem__(self, key: Hashable) -> Any:
+        hash_key = hash(key)
+        index = hash_key % self.capacity
+
+        while self.table[index]:
+            if (self.table[index][0] == key
```

Comment:
There is no need to compare hash_keys if the keys are equal
It would be the same without 'and' part